lib/private/Collaboration/Collaborators/GroupPlugin.php

```diff
@@ -106,8 +114,8 @@ public function search($search, $limit, $offset, ISearchResult $searchResult): b
 			// On page one we try if the search result has a direct hit on the
 			// user id and if so, we add that to the exact match list
 			$group = $this->groupManager->get($search);
-			if ($group instanceof IGroup && !$group->hideFromCollaboration() && (!$this->shareWithGroupOnly || in_array($group->getGID(), $userGroups))) {
-				$result['exact'][] = [
+			if ($group instanceof IGroup && !$group->hideFromCollaboration() && array_search($group->getGID(), $groupsBlockList) === false && (!$this->shareWithGroupOnly || in_array($group->getGID(), $userGroups))) {
```

Comment:
```suggestion
			if ($group instanceof IGroup
				&& !$group->hideFromCollaboration()
				&& !in_array($group->getGID(), $groupsBlockList)
				&& (!$this->shareWithGroupOnly || in_array($group->getGID(), $userGroups))
			) {
```

Why array_search instead of in_array?